File: 56-merge-intervals/56-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        
        vector<vector<int>> mergedIntervals;
        if(intervals.size() == 0) return mergedIntervals;
        
        sort(intervals.begin(), intervals.end());
        
        for(auto x : intervals){
            if(mergedIntervals.empty() or x[0] > mergedIntervals.back()[1]) mergedIntervals.push_back(x);
            else mergedIntervals.back()[1] = max(mergedIntervals.back()[1], x[1]);
        }
        return mergedIntervals; 
    }
};

File: 56-merge-intervals/56-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        merged = []
        
        intervals.sort()
        
        for x in intervals:
            if not merged or x[0] > merged[-1][1]:
                merged.append(x)
            else:
                merged[-1][1] = max(merged[-1][1], x[1])
                
        return merged
                